conans/client/rest/remote_credentials.py

```diff
@@ -32,24 +35,36 @@ def auth(self, remote, user=None, password=None):
         if user is not None and password is not None:
             return user, password
 
-        # First prioritize the cache "credentials.json" file
-        creds = self._urls.get(remote)
+        # First get the auth_plugin
+        auth_plugin = _load_auth_plugin(self.auth_plugin_path)
+        if auth_plugin is not None:
+            try:
+                plugin_user, plugin_password = auth_plugin(remote, user=user, password=password)
+            except Exception as e:
+                msg = f"Error while processing 'auth.py' plugin"
+                msg = scoped_traceback(msg, e, scope="/extensions/plugins")
+                raise ConanException(msg)
+            if plugin_user and plugin_password:
+                return plugin_user, plugin_password
+
+        # Then prioritize the cache "credentials.json" file
+        creds = self._urls.get(remote.name)
         if creds is not None:
             try:
                 return creds["user"], creds["password"]
             except KeyError as e:
                 raise ConanException(f"Authentication error, wrong credentials.json: {e}")
 
         # Then, check environment definition
-        env_user, env_passwd = self._get_env(remote, user)
+        env_user, env_passwd = self._get_env(remote.name, user)
```

Comment:
Seems like `conans/client/rest/auth_manager.py::_retry_with_new_token` is still passing a string to this function, this is what is making the tests fail

```diff
diff --git a/conans/client/rest/auth_manager.py b/conans/client/rest/auth_manager.py
index cf1ab8dde..35b5c9ddc 100644
--- a/conans/client/rest/auth_manager.py
+++ b/conans/client/rest/auth_manager.py
@@ -71,7 +71,7 @@ class ConanApiAuthManager(object):
         credentials are stored in localdb and rest method is called"""
         for _ in range(LOGIN_RETRIES):
             creds = RemoteCredentials(self._cache_folder, self._global_conf)
-            input_user, input_password = creds.auth(remote.name)
+            input_user, input_password = creds.auth(remote)
             try:
                 self._authenticate(remote, input_user, input_password)
             except AuthenticationException:
```